### PR TITLE
Textfield: required with icon, no label bug fix

### LINKED ISSUE #3209: [DatePicker]Required filed red star position is wrong without label text
### Bug Report

- __Package version(s)__: 5.10.0
- __Browser and OS versions__: 
#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? No

Requested priority: Normal

Products/sites affected: 

#### Describe the issue:

When date picker doesn't has label text, and set as required.
The red start overlaps with the calendar icon

#### Actual behavior:
image

### If applicable, please provide a codepen repro:
https://codepen.io/yudma/pen/ZXNqrK

### PR DESCRIPTION
**Pull request by lynamemi**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #3209 
- [X] Include a change request file using `$ npm run change`

#### Description of changes

- Recently made change to handle the asterisk for TextField without a label. This functionality didn't work with icons.
- Added new class + mixin to handle the same condition with icons.

**Before:**
image

**After:**
- In TextField:
image

- In DatePicker:
image

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by lynamemi** [comment]
New screenshot for textfield - improved icon quality and positioning of asterisks:
image

**Comment by lynamemi** [comment]
The screener change is intentional. Improved the icon resolution and had to adjust the positioning accordingly.

File: packages/office-ui-fabric-react/src/components/Label/LabelMixins.scss [hunk 1]
```diff
@@ -15,4 +15,15 @@
...
+}
+
+@mixin ms-Icon-is-required {
+  &::before {
+    content: ' *';
+    color: $ms-color-error;
+    position: absolute;
+    top: -4px;
+    @include ms-right(8px);
+    @include ms-padding-right(12px);
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Label/LabelMixins.scss]: this is very specific to the textfield scenario no? Why not just define it in textfield? Mixins like this make things messy. Especially now that TextField.scss have an implicit dependency on LabelMixins.scss.

File: common/changes/office-ui-fabric-react/textfield-icon-required_2017-10-24-16-24.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "comment": "New styling for required textfield with an icon but no label.",
```
Comment (dzearing) [inline common/changes/office-ui-fabric-react/textfield-icon-required_2017-10-24-16-24.json]: TextField/DatePicker: The required astrisk is now more correctly positioned.

Comment (dzearing) [reply common/changes/office-ui-fabric-react/textfield-icon-required_2017-10-24-16-24.json]: Also make sure to check variants. Are we ok with multiline?

Comment (lynamemi) [reply common/changes/office-ui-fabric-react/textfield-icon-required_2017-10-24-16-24.json]: It actually looks like the icon positioning was / is still off. I'll check in with Ben and add it to my backlog is it looks like more than a super simple fix.

**Pull request merged**